In [3]:
import pandas as pd

# looad the data set 
df=pd.read_csv("tech_salary_dataset.csv")


In [10]:
print(df.shape)
print()
df.info()

(10000, 50)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 50 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   employee_id                  10000 non-null  object 
 1   first_name                   10000 non-null  object 
 2   last_name                    10000 non-null  object 
 3   age                          10000 non-null  int64  
 4   gender                       10000 non-null  object 
 5   location                     10000 non-null  object 
 6   currency                     10000 non-null  object 
 7   primary_tech_field           10000 non-null  object 
 8   job_title                    10000 non-null  object 
 9   employment_type              10000 non-null  object 
 10  work_arrangement             10000 non-null  object 
 11  experience_years_total       10000 non-null  float64
 12  experience_years_in_field    10000 non-null  float64
 13  educ

tech_salary_dataset.csv have 1000 rows and 50 columns.



In [11]:
print(df.columns)

Index(['employee_id', 'first_name', 'last_name', 'age', 'gender', 'location',
       'currency', 'primary_tech_field', 'job_title', 'employment_type',
       'work_arrangement', 'experience_years_total',
       'experience_years_in_field', 'education_level', 'languages_spoken',
       'skills', 'primary_skill', 'num_skills', 'certifications',
       'company_size', 'annual_salary_usd', 'monthly_salary_usd',
       'hourly_rate_usd', 'annual_bonus_usd', 'bonus_percentage', 'has_equity',
       'equity_total_value_usd', 'equity_vesting_years',
       'effective_tax_rate_pct', 'annual_tax_usd', 'annual_net_salary_usd',
       'monthly_net_salary_usd', 'monthly_housing_usd',
       'monthly_transportation_usd', 'monthly_food_usd',
       'monthly_utilities_usd', 'monthly_healthcare_usd',
       'monthly_entertainment_usd', 'monthly_education_loan_usd',
       'monthly_childcare_usd', 'monthly_miscellaneous_usd',
       'monthly_total_expenses_usd', 'annual_total_expenses_usd',
       'annu

we can see all the columns of the dataset


In [13]:
df.sample(10)

,employee_id,first_name,last_name,age,gender,location,currency,primary_tech_field,job_title,employment_type,...,monthly_miscellaneous_usd,monthly_total_expenses_usd,annual_total_expenses_usd,annual_savings_usd,vacation_weeks,overtime_hours_per_month,months_since_last_promotion,num_previous_companies,job_satisfaction_score,is_actively_looking
3850,EMP-03851,Robert,Silva,35,Male,"San Francisco, USA",USD,Data Science & ML,NLP Engineer,Part-time,...,1739.20,29695.64,356347.68,15622.24,2,8.2,60,1,5,False
4790,EMP-04791,Laura,Liu,30,Female,"San Francisco, USA",USD,Mobile Development,React Native Developer,Full-time,...,504.30,11608.57,139302.84,5198.43,6,0.0,0,0,10,False
9204,EMP-09205,Ethan,Jones,38,Male,Remote (Global),USD,Frontend Development,Web Developer,Full-time,...,308.81,7389.39,88672.68,21498.91,4,10.9,55,2,7,False
5635,EMP-05636,Morgan,Anderson,38,Prefer not to say,"New York, USA",USD,Frontend Development,Frontend Developer,Full-time,...,1031.70,15224.85,182698.20,11358.35,5,8.2,2,0,10,False
8419,EMP-08420,Aisha,Choi,35,Female,"Boston, USA",USD,Frontend Development,Angular Developer,Full-time,...,1089.87,13451.31,161415.72,11668.76,4,10.2,17,3,8,False
9113,EMP-09114,Raj,Tanaka,25,Male,"Amsterdam, Netherlands",EUR,Site Reliability Engineering,Site Reliability Engineer,Full-time,...,838.64,18790.45,225485.40,22140.64,3,16.0,79,3,6,True
3195,EMP-03196,Miguel,Choi,30,Male,"Seoul, South Korea",KRW,Data Science & ML,ML Researcher,Full-time,...,1162.10,17259.87,207118.44,36370.45,4,12.5,37,1,5,False
731,EMP-00732,Avery,Moore,27,Prefer not to say,"Seattle, USA",USD,Full-Stack Development,Web Developer,Full-time,...,1244.02,22726.11,272713.32,33130.45,4,13.0,4,2,6,True
6238,EMP-06239,Mason,Yamamoto,29,Male,"Austin, USA",USD,Full-Stack Development,Senior Software Engineer,Full-time,...,1209.80,16849.99,202199.88,15664.51,3,10.2,41,3,5,False
1041,EMP-01042,Phoenix,Anderson,43,Non-Binary,"Berlin, Germany",EUR,Full-Stack Development,Senior Software Engineer,Full-time,...,488.82,9689.79,116277.48,6880.36,3,8.2,31,3,8,False


10 random sample data 

In [14]:
df.head(5)

,employee_id,first_name,last_name,age,gender,location,currency,primary_tech_field,job_title,employment_type,...,monthly_miscellaneous_usd,monthly_total_expenses_usd,annual_total_expenses_usd,annual_savings_usd,vacation_weeks,overtime_hours_per_month,months_since_last_promotion,num_previous_companies,job_satisfaction_score,is_actively_looking
0,EMP-00001,Emma,Johansson,32,Female,Remote (Global),USD,Backend Development,Platform Engineer,Full-time,...,417.44,5175.31,62103.72,6271.69,3,12.9,38,1,6,False
1,EMP-00002,Stephanie,Zhang,30,Female,"Seattle, USA",USD,QA & Testing,QA Engineer,Full-time,...,504.87,10082.84,120994.08,10151.16,4,6.5,26,1,4,False
2,EMP-00003,Raj,Lopez,32,Male,"Amsterdam, Netherlands",EUR,Backend Development,API Engineer,Full-time,...,377.88,8111.78,97341.36,4172.42,5,13.2,0,0,6,True
3,EMP-00004,Amelia,Taylor,31,Female,"London, UK",GBP,Database & Data Engineering,Analytics Engineer,Full-time,...,1140.39,23327.47,279929.64,19102.74,2,5.9,111,5,7,False
4,EMP-00005,Sophia,Gonzalez,42,Female,"Boston, USA",USD,Embedded Systems & IoT,Embedded Engineer,Full-time,...,844.73,11148.58,133782.96,3308.59,6,0.0,20,3,9,False


top 5 rows of the dataset

In [15]:
df.tail(5)

,employee_id,first_name,last_name,age,gender,location,currency,primary_tech_field,job_title,employment_type,...,monthly_miscellaneous_usd,monthly_total_expenses_usd,annual_total_expenses_usd,annual_savings_usd,vacation_weeks,overtime_hours_per_month,months_since_last_promotion,num_previous_companies,job_satisfaction_score,is_actively_looking
9995,EMP-09996,Natasha,Silva,38,Female,"Sydney, Australia",AUD,Product Management,Associate Product Manager,Full-time,...,1102.53,17158.18,205898.16,21516.12,3,7.3,52,2,7,False
9996,EMP-09997,Jacob,Chen,37,Male,"Boston, USA",USD,Full-Stack Development,Full Stack Developer,Full-time,...,833.94,14618.69,175424.28,34454.83,5,8.4,58,6,9,False
9997,EMP-09998,Kai,Muller,31,Non-Binary,"Hyderabad, India",INR,Full-Stack Development,Senior Software Engineer,Full-time,...,29.41,956.04,11472.48,2374.97,4,17.1,0,0,8,False
9998,EMP-09999,Hassan,Chia,32,Male,"Boston, USA",USD,Data Science & ML,Data Scientist,Full-time,...,1559.26,22335.50,268026.00,26685.96,4,14.3,78,2,6,False
9999,EMP-10000,Tyler,Costa,33,Male,"Seoul, South Korea",KRW,Frontend Development,UI Engineer,Part-time,...,773.86,11604.00,139248.00,12003.00,0,11.1,11,3,8,False


bottom 5 rows of the dataset

In [ ]:
df.describe()

,age,experience_years_total,experience_years_in_field,num_skills,annual_salary_usd,monthly_salary_usd,hourly_rate_usd,annual_bonus_usd,bonus_percentage,equity_total_value_usd,...,monthly_childcare_usd,monthly_miscellaneous_usd,monthly_total_expenses_usd,annual_total_expenses_usd,annual_savings_usd,vacation_weeks,overtime_hours_per_month,months_since_last_promotion,num_previous_companies,job_satisfaction_score
count,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.00000,...,10000.000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000
mean,31.637600,5.717070,4.019440,5.494200,171786.241744,14315.520181,91.601258,12363.473986,7.214300,28710.88860,...,916.206,818.562067,14148.267573,169779.210876,16530.105659,3.561300,10.581700,34.515400,1.629600,6.974100
std,6.856892,3.689415,2.823968,1.716176,72550.937998,6045.911530,45.335040,10863.881829,5.159558,100995.44362,...,1062.339,477.180884,7668.022002,92016.264021,10854.098074,1.543464,7.590467,32.959976,1.539364,2.157428
min,18.000000,0.500000,0.500000,3.000000,19846.270000,1653.860000,9.540000,0.000000,0.000000,0.00000,...,0.000,20.650000,424.070000,5088.840000,-4103.200000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,27.000000,2.800000,1.800000,4.000000,121355.045000,10112.925000,61.190000,2282.800000,5.000000,0.00000,...,0.000,497.502500,9118.967500,109427.610000,8731.762500,2.000000,4.500000,8.000000,0.000000,6.000000
50%,32.000000,5.500000,3.600000,5.000000,167055.180000,13921.265000,85.295000,11302.280000,8.000000,0.00000,...,0.000,756.215000,13336.920000,160043.040000,14228.585000,4.000000,10.100000,26.000000,1.000000,7.000000
75%,36.000000,8.200000,5.700000,7.000000,216939.867500,18078.325000,113.770000,18781.787500,10.000000,9773.00000,...,1800.000,1087.377500,18536.777500,222441.330000,21807.995000,5.000000,15.700000,52.000000,2.000000,9.000000
max,58.000000,19.600000,17.800000,8.000000,479033.820000,39919.490000,373.060000,81019.780000,20.000000,995997.00000,...,3600.000,3868.360000,52337.650000,628051.800000,86728.360000,6.000000,54.100000,224.000000,10.000000,10.000000


this is the information of 10000 employess

Average

average age= 31.637600, total average years of experience=5.717070 , average year of experience in current field=4.019440 , average number of skill=5.494200 , average annual salary= 171786.241744, average monthly salary= 14315.520181, average hourly rate=91.601258 , average annual bonus= 12363.473986, average bonus percentage=7.214300 , average annual expenses=169779.210876 , average annaul saving=16530.105659 , average vacation per week=3.561300 , average overtime hours per month= 10.581700,

Minimum

minimum age= 18.0, total minimum years of experience= 0.500, minimum year of experience in current field=0.50, minimum number of skill= 3.0, minimum annual salary= 19846.270000, minimum monthly salary=1653.860000 , minimum hourly rate=9.540000 , minimum annual bonus= 0, minimum bonus percentage=0 , minimum annual expenses= 5088.840000, minimum annaul saving=-4103.200000, minimum vacation per week=0.0 , minimum overtime hours per month=0 ,

Maximum

maximum age=58.0 , total maximum years of experience= 19.600, maximum year of experience in current field= 17.800, maximum number of skill= 8.00, maximum annual salary= 479033.820000, maximum monthly salary=39919.490000 , maximum hourly rate= 373.060000, maximum annual bonus= 81019.780000, maximum bonus percentage=20.0 , maximum annual expenses=628051.80 , maximum annaul saving=86728.360000, maximum vacation per week= 6.0, maximum overtime hours per month=54.100000,

In [19]:
df.describe(include='object')

,employee_id,first_name,last_name,gender,location,currency,primary_tech_field,job_title,employment_type,work_arrangement,education_level,languages_spoken,skills,primary_skill,certifications,company_size
count,10000,10000,10000,10000,10000,10000,10000,10000,10000,10000,10000,10000,10000,10000,5492,10000
unique,10000,110,74,4,19,10,14,78,4,4,6,1302,9915,133,677,5
top,EMP-00001,Alex,Novak,Male,"San Francisco, USA",USD,Full-Stack Development,Platform Engineer,Full-time,Remote,Bachelor's Degree,English,TypeScript; Redis; Docker,Python,TensorFlow Developer Certificate,Enterprise (1000+)
freq,1,185,173,5504,1280,5417,1233,363,8011,3003,4038,3455,3,377,280,2047


In [20]:
df.describe(include='all')

,employee_id,first_name,last_name,age,gender,location,currency,primary_tech_field,job_title,employment_type,...,monthly_miscellaneous_usd,monthly_total_expenses_usd,annual_total_expenses_usd,annual_savings_usd,vacation_weeks,overtime_hours_per_month,months_since_last_promotion,num_previous_companies,job_satisfaction_score,is_actively_looking
count,10000,10000,10000,10000.000000,10000,10000,10000,10000,10000,10000,...,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000
unique,10000,110,74,NaN,4,19,10,14,78,4,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2
top,EMP-00001,Alex,Novak,NaN,Male,"San Francisco, USA",USD,Full-Stack Development,Platform Engineer,Full-time,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,False
freq,1,185,173,NaN,5504,1280,5417,1233,363,8011,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,7994
mean,NaN,NaN,NaN,31.637600,NaN,NaN,NaN,NaN,NaN,NaN,...,818.562067,14148.267573,169779.210876,16530.105659,3.561300,10.581700,34.515400,1.629600,6.974100,NaN
std,NaN,NaN,NaN,6.856892,NaN,NaN,NaN,NaN,NaN,NaN,...,477.180884,7668.022002,92016.264021,10854.098074,1.543464,7.590467,32.959976,1.539364,2.157428,NaN
min,NaN,NaN,NaN,18.000000,NaN,NaN,NaN,NaN,NaN,NaN,...,20.650000,424.070000,5088.840000,-4103.200000,0.000000,0.000000,0.000000,0.000000,0.000000,NaN
25%,NaN,NaN,NaN,27.000000,NaN,NaN,NaN,NaN,NaN,NaN,...,497.502500,9118.967500,109427.610000,8731.762500,2.000000,4.500000,8.000000,0.000000,6.000000,NaN
50%,NaN,NaN,NaN,32.000000,NaN,NaN,NaN,NaN,NaN,NaN,...,756.215000,13336.920000,160043.040000,14228.585000,4.000000,10.100000,26.000000,1.000000,7.000000,NaN
75%,NaN,NaN,NaN,36.000000,NaN,NaN,NaN,NaN,NaN,NaN,...,1087.377500,18536.777500,222441.330000,21807.995000,5.000000,15.700000,52.000000,2.000000,9.000000,NaN


In [21]:
df.isnull().sum()

employee_id                       0
first_name                        0
last_name                         0
age                               0
gender                            0
location                          0
currency                          0
primary_tech_field                0
job_title                         0
employment_type                   0
work_arrangement                  0
experience_years_total            0
experience_years_in_field         0
education_level                   0
languages_spoken                  0
skills                            0
primary_skill                     0
num_skills                        0
certifications                 4508
company_size                      0
annual_salary_usd                 0
monthly_salary_usd                0
hourly_rate_usd                   0
annual_bonus_usd                  0
bonus_percentage                  0
has_equity                        0
equity_total_value_usd            0
equity_vesting_years        

In [30]:
df['certifications'].unique()

array([nan, 'AWS Certified Solutions Architect',
       'Linux Foundation Certified', 'Google Cloud Professional',
       'Azure Solutions Architect; CompTIA Security+; PMP',
       'AWS Certified Developer', 'Kubernetes CKA', 'PMP; CEH', 'CEH',
       'CISSP', 'CEH; PMP',
       'Azure Solutions Architect; TensorFlow Developer Certificate',
       'CompTIA Security+; AWS Certified Developer',
       'AWS Certified Solutions Architect; PMP',
       'Certified Scrum Master; Azure Solutions Architect',
       'Kubernetes CKA; CEH; Azure Solutions Architect',
       'CEH; Kubernetes CKA; Certified Scrum Master',
       'TensorFlow Developer Certificate', 'Azure Solutions Architect',
       'CEH; AWS Certified Developer', 'CompTIA Security+',
       'CEH; Linux Foundation Certified',
       'Kubernetes CKA; Linux Foundation Certified',
       'TensorFlow Developer Certificate; Google Cloud Professional; Linux Foundation Certified',
       'CEH; AWS Certified Developer; PMP',
       'AWS Ce

In [31]:
df['certifications'].value_counts()

certifications
TensorFlow Developer Certificate                                                       280
CEH                                                                                    277
Google Cloud Professional                                                              273
AWS Certified Developer                                                                272
PMP                                                                                    269
                                                                                      ... 
PMP; AWS Certified Developer; CompTIA Security+                                          1
CEH; Linux Foundation Certified; CISSP                                                   1
TensorFlow Developer Certificate; AWS Certified Solutions Architect; Kubernetes CKA      1
AWS Certified Solutions Architect; Certified Scrum Master; PMP                           1
CISSP; Linux Foundation Certified; Certified Scrum Master                  

In [33]:
df['certifications'].mode()

0    TensorFlow Developer Certificate
Name: certifications, dtype: object

In [34]:
df['annual_salary_usd'].max()

479033.82

In [42]:
df['certifications']=df['certifications'].fillna('No certification')

In [46]:
df['certifications'].value_counts()
df_cleaned=df

In [49]:
df_cleaned['certifications'].mode()

0    No certification
Name: certifications, dtype: object

In [50]:
df_cleaned.shape

(10000, 50)

In [51]:
df_cleaned['annual_salary_usd']

0       100290.66
1        92383.12
2       107153.59
3       217074.54
4       152246.34
          ...    
9995    218700.43
9996    170835.76
9997     25695.15
9998    264531.10
9999    163072.71
Name: annual_salary_usd, Length: 10000, dtype: float64

In [53]:
max_salary=df_cleaned['annual_salary_usd'].max()
print(max_salary)

479033.82


In [54]:
df_cleaned['annual_salary_usd']==max_salary

0       False
1       False
2       False
3       False
4       False
        ...  
9995    False
9996    False
9997    False
9998    False
9999    False
Name: annual_salary_usd, Length: 10000, dtype: bool

In [55]:
df_cleaned[df_cleaned['annual_salary_usd']==max_salary]

,employee_id,first_name,last_name,age,gender,location,currency,primary_tech_field,job_title,employment_type,...,monthly_miscellaneous_usd,monthly_total_expenses_usd,annual_total_expenses_usd,annual_savings_usd,vacation_weeks,overtime_hours_per_month,months_since_last_promotion,num_previous_companies,job_satisfaction_score,is_actively_looking
3152,EMP-03153,Lars,Williams,35,Male,"San Francisco, USA",USD,Product Management,Product Manager,Full-time,...,3868.36,43783.88,525406.56,51177.04,4,11.3,45,2,7,False


to get maximum salary must be male of above 30 years, should be in the post of manager, employment type must be full_time.

In [56]:
df_cleaned['job_title'].value_counts()

job_title
Platform Engineer           363
Web Developer               363
Senior Software Engineer    256
Full Stack Developer        252
Software Engineer           241
                           ... 
Penetration Tester           61
Graphics Programmer          61
Smart Contract Engineer      60
Protocol Engineer            56
Web3 Developer               53
Name: count, Length: 78, dtype: int64

In [ ]:
# most common jon title
most_common_job_title="Platform Engineer"
print(f"most common job title is {most_common_job_title},{df_cleaned['job_title'].value_counts().max()}")

most common job title is Plateform Engineer,363


In [69]:
most_common_job_title="Platform Engineer"
Platform_Engineer =df_cleaned[df_cleaned['job_title']==most_common_job_title]
Platform_Engineer

,employee_id,first_name,last_name,age,gender,location,currency,primary_tech_field,job_title,employment_type,...,monthly_miscellaneous_usd,monthly_total_expenses_usd,annual_total_expenses_usd,annual_savings_usd,vacation_weeks,overtime_hours_per_month,months_since_last_promotion,num_previous_companies,job_satisfaction_score,is_actively_looking
0,EMP-00001,Emma,Johansson,32,Female,Remote (Global),USD,Backend Development,Platform Engineer,Full-time,...,417.44,5175.31,62103.72,6271.69,3,12.9,38,1,6,False
15,EMP-00016,David,Ali,32,Male,"London, UK",GBP,DevOps & Cloud,Platform Engineer,Contract,...,668.29,14561.65,174739.80,13903.21,1,15.6,9,0,8,False
76,EMP-00077,Lisa,Liu,27,Female,"Boston, USA",USD,Backend Development,Platform Engineer,Full-time,...,989.93,17345.38,208144.56,28481.22,4,0.0,94,4,2,False
83,EMP-00084,David,Tran,31,Male,"Tel Aviv, Israel",ILS,Backend Development,Platform Engineer,Full-time,...,644.73,10108.22,121298.64,12021.80,5,19.0,14,1,10,False
97,EMP-00098,Aaron,Malik,29,Male,"Amsterdam, Netherlands",EUR,DevOps & Cloud,Platform Engineer,Full-time,...,760.89,12711.60,152539.20,17053.16,4,1.8,6,0,7,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
9756,EMP-09757,Alex,Ng,37,Male,Singapore,SGD,Backend Development,Platform Engineer,Full-time,...,781.58,12481.75,149781.00,16529.63,5,7.6,10,1,8,False
9823,EMP-09824,Alex,Andersen,36,Male,"Miami, USA",USD,Backend Development,Platform Engineer,Full-time,...,401.33,9040.13,108481.56,15467.30,4,5.6,21,2,7,False
9929,EMP-09930,Donna,Khan,43,Female,"New York, USA",USD,DevOps & Cloud,Platform Engineer,Full-time,...,1171.89,15385.91,184630.92,40126.25,3,23.0,4,3,10,True
9939,EMP-09940,William,Fernandes,44,Male,"Bangalore, India",INR,Backend Development,Platform Engineer,Full-time,...,49.54,1470.15,17641.80,4291.93,5,0.0,0,0,6,True


In [73]:
PE_max_salary=Platform_Engineer['annual_salary_usd'].max()
print(PE_max_salary)
top_earner=Platform_Engineer[Platform_Engineer['annual_salary_usd']==PE_max_salary]
print(top_earner)

392294.58
     employee_id first_name last_name  age  gender            location  \
3631   EMP-03632       Lisa     Malik   36  Female  San Francisco, USA   

     currency primary_tech_field          job_title employment_type  ...  \
3631      USD     DevOps & Cloud  Platform Engineer       Full-time  ...   

     monthly_miscellaneous_usd  monthly_total_expenses_usd  \
3631                   2925.78                    42386.58   

      annual_total_expenses_usd annual_savings_usd vacation_weeks  \
3631                  508638.96           42206.89              3   

     overtime_hours_per_month months_since_last_promotion  \
3631                      9.8                          47   

      num_previous_companies job_satisfaction_score is_actively_looking  
3631                       3                      2               False  

[1 rows x 50 columns]


In [75]:
oldest_person=df_cleaned['age'].max()
print(oldest_person)
olderst_employee=df_cleaned[df_cleaned['age']==oldest_person]
print(olderst_employee)

58
    employee_id first_name  last_name  age gender          location currency  \
660   EMP-00661   Mohammed  Hernandez   58   Male  Bangalore, India      INR   

    primary_tech_field                job_title employment_type  ...  \
660  Blockchain & Web3  Smart Contract Engineer       Full-time  ...   

    monthly_miscellaneous_usd  monthly_total_expenses_usd  \
660                     56.38                     2332.08   

     annual_total_expenses_usd annual_savings_usd vacation_weeks  \
660                   27984.96            6896.72              3   

    overtime_hours_per_month months_since_last_promotion  \
660                      8.2                           3   

     num_previous_companies job_satisfaction_score is_actively_looking  
660                       0                      3               False  

[1 rows x 50 columns]


In [87]:
# average salary by gender
avg_salary_by_gender=df_cleaned.groupby('gender')['annual_salary_usd'].max()
avg_salary_by_gender


gender
Female               475053.71
Male                 479033.82
Non-Binary           465540.18
Prefer not to say    406586.19
Name: annual_salary_usd, dtype: float64

In [89]:
df_cleaned['education_level'].value_counts()


education_level
Bachelor's Degree     4038
Master's Degree       2488
Bootcamp              2197
PhD                    785
Associate's Degree     304
High School            188
Name: count, dtype: int64

In [96]:
bachelor_degree_people=df_cleaned[df_cleaned['education_level']=="Bachelor's Degree"]
print(bachelor_degree_people)

     employee_id first_name last_name  age      gender  \
2      EMP-00003        Raj     Lopez   32        Male   
5      EMP-00006      Kenji     Davis   28        Male   
6      EMP-00007      Logan       Lim   29        Male   
9      EMP-00010     Dmitri     Patel   36        Male   
10     EMP-00011      Freya     Costa   36      Female   
...          ...        ...       ...  ...         ...   
9984   EMP-09985     Dmitri     Malik   26        Male   
9989   EMP-09990      Aisha  Andersen   22      Female   
9990   EMP-09991      Logan      Wang   40        Male   
9992   EMP-09993     Justin     Costa   35        Male   
9997   EMP-09998        Kai    Muller   31  Non-Binary   

                    location currency            primary_tech_field  \
2     Amsterdam, Netherlands      EUR           Backend Development   
5         San Francisco, USA      USD              Game Development   
6              New York, USA      USD  Site Reliability Engineering   
9              New 

In [98]:
# find the age between 30-40
import numpy as np
age_30_40=df_cleaned[np.logical_and((df_cleaned['age']>=30), (df_cleaned['age']<=40))]
print(age_30_40)
age_30_40.shape

     employee_id first_name  last_name  age             gender  \
0      EMP-00001       Emma  Johansson   32             Female   
1      EMP-00002  Stephanie      Zhang   30             Female   
2      EMP-00003        Raj      Lopez   32               Male   
3      EMP-00004     Amelia     Taylor   31             Female   
7      EMP-00008     Jordan       Sato   32  Prefer not to say   
...          ...        ...        ...  ...                ...   
9995   EMP-09996    Natasha      Silva   38             Female   
9996   EMP-09997      Jacob       Chen   37               Male   
9997   EMP-09998        Kai     Muller   31         Non-Binary   
9998   EMP-09999     Hassan       Chia   32               Male   
9999   EMP-10000      Tyler      Costa   33               Male   

                    location currency           primary_tech_field  \
0            Remote (Global)      USD          Backend Development   
1               Seattle, USA      USD                 QA & Testing 

(5106, 50)

In [ ]:
# filter job title  aged between 30 to 40
abc=df_cleaned[df_cleaned['age'].between(30,40)][['job_title','age','first_name','location']]
abc

,job_title,age,first_name,location
0,Platform Engineer,32,Emma,Remote (Global)
1,QA Engineer,30,Stephanie,"Seattle, USA"
2,API Engineer,32,Raj,"Amsterdam, Netherlands"
3,Analytics Engineer,31,Amelia,"London, UK"
7,Software Engineer,32,Jordan,"Miami, USA"
...,...,...,...,...
9995,Associate Product Manager,38,Natasha,"Sydney, Australia"
9996,Full Stack Developer,37,Jacob,"Boston, USA"
9997,Senior Software Engineer,31,Kai,"Hyderabad, India"
9998,Data Scientist,32,Hassan,"Boston, USA"


In [ ]:
# filter salary between 300000 to 400000
xyz=df_cleaned[df_cleaned['annual_salary_usd'].between(300000,400000)][['job_title','age','first_name','location']]
xyz

,job_title,age,first_name,location
27,iOS Developer,27,Matthew,"New York, USA"
42,Technical Product Manager,25,Jacob,"London, UK"
57,Web Developer,28,Mohammed,"San Francisco, USA"
115,SRE,40,Joseph,"San Francisco, USA"
131,Computer Vision Engineer,34,Alex,"London, UK"
...,...,...,...,...
9917,Technical Product Manager,45,Ali,"New York, USA"
9934,Product Manager,23,Michael,"New York, USA"
9935,SRE,25,Sanjay,"San Francisco, USA"
9961,ML Researcher,36,Avery,"London, UK"
